Release 1.4.0 into trunk

AliSoftware opened the pull request:
New version 1.4.0. Be sure to create a GitHub Release and tag once this PR gets merged.

jkmassel committed:
Fix a bug in the `plugin_root` method

When generating promo screenshots, the error `Unable to determine the plugin root directory` would come up – this fixes it by better determining the location of the plugin root in production environments (where the .gemspec file would not be present in `fastlane-plugin-wpmreleasetoolkit`.

Make rubocop happy

Add release note

Make rubocopy happy

jkmassel committed:
Fix CircleCI issue

loremattei committed:
Merge pull request #286 from wordpress-mobile/release/1.3.1

Merge Release/1.3.1 into develop

jkmassel committed:
Update dependencies

jkmassel committed:
Merge branch 'develop' into fix/broken-plugin_root-for-promo-screenshots

mokagio committed:
Remove comments and fix formatting for ios_bump_version_hotfix configs

The comments were unnecessary. Nothing but the template helper comments.

mokagio committed:
Add skip_deliver parameter to ios_bump_version_hotfix

mokagio committed:
Only update Deliverfile in ios_bump_version_hotfix if skip_deliver false

AliSoftware committed:
Fix hotfix actions

AliSoftware committed:
Fix incorrect var used

AliSoftware committed:
Avoid get/set of alpha version since it doesn't change

AliSoftware committed:
Make rubocop happy

AliSoftware committed:
Add CHANGELOG entry

jkmassel committed:
Merge pull request #282 from wordpress-mobile/update/bigdecimal

Update dependencies

jkmassel committed:
Update lib/fastlane/plugin/wpmreleasetoolkit/helper/filesystem_helper.rb

Co-authored-by: Olivier Halligon <olivier.halligon@automattic.com>

jkmassel committed:
Make directory crawling a tiny bit safer

jkmassel committed:
Merge pull request #276 from wordpress-mobile/fix/broken-plugin_root-for-promo-screenshots

Fix a bug in the `plugin_root` method

AliSoftware committed:
Merge branch 'develop' into fix/hotfix-actions

Conflicts:
  CHANGELOG.md

AliSoftware committed:
Merge pull request #288 from wordpress-mobile/fix/hotfix-actions

Fix hotfix actions

mokagio committed:
Revert "Update dependencies"

This reverts commit 1722c13799004064312d3b283a5f77419c38bd2d.

See https://github.com/wordpress-mobile/release-toolkit/issues/289

mokagio committed:
Track result of calling `bundle install` after previous commit

mokagio committed:
Add CocoaPods as a development dependency

This should help avoid the dependency resolution issue encountered in #289.

mokagio committed:
Update CHANGELOG adding entry for #287

AliSoftware committed:
Merge pull request #290 from wordpress-mobile/revert-activesearch-v6-bump

Revert activesupport version bump

AliSoftware committed:
Merge pull request #287 from wordpress-mobile/add-skip-deliver-in-ios-hotfix-version-bump

Add option to skip updating Deliverfile to ios_bump_version_hotfix

AliSoftware committed:
Bumped to version 1.4.0

mokagio commented:
I left a comment regarding the failing GitHub Actions build [here](https://github.com/wordpress-mobile/release-toolkit/pull/292#issuecomment-891758279).

mokagio committed:
Disable platform specific gem installs via Bundler

See discussion here: https://github.com/wordpress-mobile/release-toolkit/pull/292#issuecomment-891758279

To ensure the setting is persisted for every user of the repository, I
decided to track the Bundler config file. This seems like a good idea
anyway, even though Bundler currently has a bug resulting in that file
being changed sometimes, which can be confusing for developers.

See https://github.com/rubygems/rubygems/issues/3708

mokagio committed:
Skip test gems group install in GitHub Actions workflow

We don't need them because we don't run the tests in that workflow.

mokagio committed:
Remove no-op option `--with screenshots` from CircleCI `bundle install`

There is no "screenshots" group. That option didn't produce any effect.

mokagio committed:
Add CHANGELOG entry for #293

mokagio commented:
Opened #293 to address the CI issue.

mokagio reviewed (approved):
LGTM. 

Once #293 gets into the release branch, we should be able to merge and ship.

AliSoftware committed:
Merge pull request #293 from wordpress-mobile/disable-platform-specific-gems

AliSoftware commented on CHANGELOG.md:
```diff
@@ -28,6 +39,9 @@ _None_
 
+### Internal Changes
+
+* Opt-out from installing platform-specific gems with Bundler [#293]
```
This is in the wrong place, should be in 1.4.0 not 1.3.1

AliSoftware committed:
Move 293 entry of CHANGELOG in right section

AliSoftware committed:
bundle update

AliSoftware committed:
Add CHANGELOG entry

AliSoftware committed:
Merge pull request #294 from wordpress-mobile/bundle-update

Fix security issues with old gems
